src/main/java/com/atlassian/jira/cloud/jenkins/deploymentinfo/service/ChangeLogIssueKeyExtractor.java

```diff
@@ -49,4 +67,13 @@
                 .map(IssueKey::toString)
                 .collect(Collectors.toSet());
     }
+
+    private boolean isBuildSuccessful(@CheckForNull final WorkflowRun workflowRun) {
+
+        return Optional.ofNullable(workflowRun)
+                .map(Run::getResult)
+                .map(Result::toString)
+                .map(Result.SUCCESS.toString()::equals)
+                .orElse(true);
```

Comment:
@nsurendran1991 thanks for the review.

* **Scenario 1 - The `workflowRun` parameter is null.**
In this case the method returns `true`, which means the keys from the "previous" build WILL NOT be extracted.

* **Scenario 2 - The result of the `workflowRun` is null.** 
I actually don't really know if this is possible at all. In this case the method returns `true` and the keys from the "previous" build WILL NOT be extracted. The outcome of this scenario can be a subject for discussion.

* **Scenario 3 - The result is `SUCCESS`.** 
The method returns `true` and the keys from the "previous" build WILL NOT be extracted.

* **Scenario 4 - The result is NOT `SUCCESS`.** 
Only in this case the method returns `false` and the keys from the "previous" build WILL be extracted.

Maybe this method should be renamed to something like "isBuildForExtract" and negate the result inside since checking if build is successful is not the exact responsibility. The intention of the method can be more understandable afterwords.

Reply:
So the intention for the entire change is to retrieve issue keys from last build failure?

Reply:
Yes. From all direct previous builds without the successful one.